File: python/lc13.py
class Solution:
    def romanToInt(self, s: str) -> int:
        m = {'I':1,'V':5,'X':10,'L':50,'C':100,'D':500, 'M':1000}
        ans = 0
        prev = 4000
        if len(s) == 1:
            return m[s]
        for i in range(1,len(s)):
            cur = m[s[i-1]]
            nxt = m[s[i]]
            if nxt > cur:
                ans -= cur
            else:
                ans += cur

        return ans + m[s[-1]]

        

File: cs/lc13.cs
public class Solution {
    public int RomanToInt(string s) {
        var m = new Dictionary<char,int>();
        m.Add('I',1);
        m.Add('V',5);
        m.Add('X',10);
        m.Add('L',50);
        m.Add('C',100);
        m.Add('D',500);
        m.Add('M',1000);
        int ans = 0;

        for (int i =1;i< s.Length;i++)
        {
            int cur = m[s[i-1]];
            int nxt = m[s[i]];
            if (nxt > cur)
                ans -= cur;
            else
                ans += cur;
        }
        return ans + m[s[s.Length-1]];
    }
}
